### DISCUSSION TITLE
Application launched through PIX throws a stack overflow exception when creating SDL window

### DISCUSSION THREAD

**Discussion by Marlax0** [amerkoleci/Alimer.Bindings.SDL#3] [Q&A] [answered]
I don't know what's happening here.

I can run my application normally. I can even capture it through RenderDoc.
But if I try to launch it for GPU capture with PIX, it silently crashes as soon as  `SDL_CreateWindow` (or any other variant) is called.

I had to check the Windows Event Viewer to see it was a stack overflow exception.
`Exception code: 0xc00000fd`

I don't understand the code here well enough to try to debug this on my own.
It looks like `SDL_CreateWindow` just recursively calls itself forever, but that can't be right. But also that's seemingly happening?
It doesn't make sense why this is only happening through PIX.

**Comment by amerkoleci** [comment]
I transferred it as discussion as it's directly an issue.

Can you share a repro project please?

**Reply by Marlax0** [reply]
I can upload one but this can be done in a few lines of code.
[SDLTests.zip](https://github.com/amerkoleci/Alimer.Bindings.SDL/files/13288523/SDLTests.zip)

```
public static void Main()
{
    if (SDL_Init(SDL_InitFlags.Timer) != 0)
    {
        throw new InvalidOperationException(SDL_GetErrorString());
    }

    Console.WriteLine("Creating SDL window...");

    SDL_Window window = SDL_CreateWindow("Test", 1280, 720, SDL_WindowFlags.Resizable);

    Console.WriteLine("SDL window created.");

    SDL_DestroyWindow(window);

    SDL_Quit();

    Console.ReadKey();
}
```
If I run it through PIX for GPU capture, I don't see the "SDL window created." message. It crashes.
>Faulting application name: SDLTests.exe, version: 1.0.0.0, time stamp: 0x653067fb
Faulting module name: ntdll.dll, version: 10.0.19041.3393, time stamp: 0xfeef31d3
Exception code: 0xc00000fd
Fault offset: 0x000000000001998e
Faulting process id: 0x94fc
Faulting application start time: 0x01da11b678e4512e
Faulting application path: D:\Projects\Visual Studio Projects\SDLTests\SDLTests\bin\Release\net7.0\SDLTests.exe
Faulting module path: C:\WINDOWS\SYSTEM32\ntdll.dll
Report Id: 5e2e1d2d-60c8-4fc3-8e87-6aad522c7f3e
Faulting package full name: 
Faulting package-relative application ID:

**Reply by amerkoleci** [reply]
Not sure what you want to capture with PIX? you need to create a D3D12 or D3D11 device

**Reply by Marlax0** [reply]
I have a main project that has all that. PIX never gets to that point.
This is the minimum code required to replicate the issue.

**Reply by amerkoleci** [reply]
I see, it could be PIX issue as well, will try tomorrow and see how it behaves on my PC

**Reply by Marlax0** [reply]
Just to note: I'm only using SDL for windowing. I pass the window off to my own D3D12 graphics system.
Perhaps I'm missing some flag or some other config setting. SDL3 is still new to me.

**Reply by amerkoleci** [reply]
You need to get win32 handle from sdl window and pass that to DXGI SwapChain

**Reply by Marlax0** [reply]
Yeah I do that, and like I said, everything works as expected, _except_ when launched through PIX.
Here's a screenshot of my application working normally.
![image](https://github.com/amerkoleci/Alimer.Bindings.SDL/assets/42450099/df4d8ae6-bee4-4ffd-bff5-3d0fe35a75be)
Possibly SDL3 is new enough this is just an SDL3 with PIX error, but I can't find anything on that.

**Reply by Marlax0** [reply, marked as answer]
I spoke to someone working on PIX, and they believe it's an SDL3 problem.
[They've filed an issue over there.](https://github.com/libsdl-org/SDL/issues/8519)

Thank you for your time, and sorry for the confusion.
